mlflow/tracking/fluent.py

```diff
@@ -146,6 +147,13 @@ def start_run(run_id=None, experiment_id=None, run_name=None, nested=False):
         if run_name is not None:
             user_specified_tags[MLFLOW_RUN_NAME] = run_name
 
+        if _active_run_stack:
+            user_specified_tags[MLFLOW_ROOT_RUN_ID] = _active_run_stack[0].data.tags[MLFLOW_ROOT_RUN_ID]
+        else:
+            user_specified_tags[MLFLOW_ROOT_RUN_ID] = uuid.uuid4().hex
+
+
+
         tags = context_registry.resolve_tags(user_specified_tags)
 
         active_run_obj = MlflowClient().create_run(
```

Comment:
MlflowClient().set_tag(run_id=active_run_obj.info.run_id, key=MLFLOW_ROOT_RUN_ID, value=active_run_obj.info.run_id)

Reply:
Not needed, since second RPC is not made